app/broadcast_message/rest.py

```diff
@@ -0,0 +1,123 @@
+from datetime import datetime
+
+import iso8601
+from flask import Blueprint, jsonify, request
+
+from app.dao.templates_dao import dao_get_template_by_id_and_service_id
+from app.dao.users_dao import get_user_by_id
+from app.dao.broadcast_message_dao import (
+    dao_create_broadcast_message,
+    dao_get_broadcast_message_by_id_and_service_id,
+    dao_update_broadcast_message,
+)
+from app.dao.services_dao import dao_fetch_service_by_id
+from app.errors import register_errors
+from app.models import BroadcastMessage, BroadcastStatusType
+from app.broadcast_message.broadcast_message_schema import (
+    create_broadcast_message_schema,
+    update_broadcast_message_schema,
+    update_broadcast_message_status_schema,
+)
+from app.schema_validation import validate
+
+broadcast_message_blueprint = Blueprint(
+    'broadcast_message',
+    __name__,
+    url_prefix='/service/<uuid:service_id>/broadcast-message'
+)
+register_errors(broadcast_message_blueprint)
+
+
+def _parse_nullable_datetime(dt):
+    if dt:
+        return iso8601.parse_date(dt).replace(tzinfo=None)
+    return dt
+
+
+@broadcast_message_blueprint.route('', methods=['GET'])
+def get_broadcast_messages_for_service(service_id):
+    service = dao_fetch_service_by_id(service_id)
+    # TODO: should this return template content/data in some way? or can we rely on them being cached admin side.
+    # we might need stuff like template name for showing on the dashboard.
+    # TODO: should this paginate or filter on dates or anything?
```

Comment:
I think the dashboard should work like this:
- waiting for approval (as many as exist – no pagination)
- live (as many as exist – no pagination)
- previous (up to 50, then pagination)

So if there are more than 50 previous broadcasts then clicking ‘next →‘ will take you to page 2, which only has previous broadcasts.

This could either work by:
- admin app makes 3 calls to the API
- or admin app makes 1 call and API responds with a `dict` that has 3 keys

I’m happy to do either, or make no changes here for now and let the admin app do grouping but not pagination.

Reply:
i'm gonna do nothing for now because we don't have 1 broadcast message, yet alone 50. but at some point it should start ordering by .... `updated_at` i think. not sure whether admin making three calls or the response being richer is preferable, but i think the admin making three calls is simpler to understand maybe?

Reply:
👍 to doing nothing for now